### PR TITLE
Update process-agent to be able to do profiling

### PR DESCRIPTION
**Pull request by shang-wang**
### What does this PR do?

This PR adds the ability for process-agent to do profiling. Note that the enable/disable switch is `process_config.profiling.enabled`, but the profiling configurations are read from the main agent config.

cc: @DataDog/processes 

### Motivation

Enable process-agent to do profiling if needed.

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -876,6 +876,24 @@ api_key:
   #   - 'sql*'
   #   - '*pass*d*'
 
+{{- if .Profiling -}}
+  ## @param profiling - custom object - optional
+  ## Enter specific configurations for profiling.
+  ## Uncomment this parameter and the one below to enable profiling.
+  ## See https://docs.datadoghq.com/tracing/profiling/
+  #
+  ## NOTE: If enabled this option may incur billing charges or other
```
Comment (sarina-dd) [inline pkg/config/config_template.yaml:885]:
```suggestion
  ## NOTE: If enabled this option may impact your billing, or enable
```

File: pkg/config/config_template.yaml [hunk 2]
```diff
@@ -876,6 +876,24 @@ api_key:
...
+  ##       unexpected side-effects (ie. process-agent profiles showing with your
```
Comment (sarina-dd) [inline pkg/config/config_template.yaml:886]:
```suggestion
  ##      other features (e.g. process-agent profiles showing with your
```

**Review by sarina-dd — DISMISSED** [review]

**Review by robertjli — APPROVED** [review]

File: pkg/config/config_template.yaml (hunk 1 above)
Comment (shang-wang) [reply pkg/config/config_template.yaml]: @sarina-dd I updated this, can you take another look please?

File: pkg/config/config_template.yaml (hunk 2 above)
Comment (shang-wang) [reply pkg/config/config_template.yaml]: @sarina-dd I updated this, can you take another look please?

**Review by apigirl — APPROVED** [review]
docs look good!

**Review by hush-hush — APPROVED** [review]

**Pull request merged**